dask/array/fft.py

```diff
@@ -28,31 +30,37 @@ def _fft_out_chunks(a, n, axis):
     return chunks
 
 
-def _rfft_out_chunks(a, n, axis):
+def _rfft_out_chunks(a, n, axis, rtype):
     if n is None:
         n = a.chunks[axis][0]
     chunks = list(a.chunks)
-    chunks[axis] = (n // 2 + 1,)
+    if issubclass(np.dtype(rtype).type, numbers.Real):
+        chunks[axis] = (n,)
+    else:
+        chunks[axis] = (n // 2 + 1,)
```

Comment:
I don't actually care about this specific situation of complex dtypes.  My question was intended to reinforce this main point: 

> My question is "are we using dtype to infer which module this is coming from?" If so then we should probably just condition on the module itself rather than back out which module is being used based on what is happening to the dtype

I'm ok not supporting scipy functions.  But we need to actively err here, not just avoid testing them.

Generally the goal of this PR appears to be "allow people to come in with their own fft modules".  I guess that the scipy case shows that we can't assume that the "chunk" functions are general.